src/edu/csus/ecs/pc2/core/model/Problem.java

```diff
@@ -2002,4 +2048,131 @@ public void setVivaInputValidatorHasBeenRun(boolean vivaInputValidatorHasBeenRun
+     * {@link #getSandboxType()} is something other than {@link SandboxType#NONE}.
+     * 
+     * @return the currently-defined sandbox program name.
+     */
+    public String getSandboxProgramName() {
+        if (sandboxProgramName == null) {
+            sandboxProgramName = Constants.PC2_INTERNAL_SANDBOX_PROGRAM_NAME;
+        }
+        return sandboxProgramName;
+    }
+
+    /**
+     * Sets the name of the sandbox program used by this Problem.
+     * Note that setting a sandbox program name does NOT in and of itself cause the specified sandbox to be
+     * used; the Admin must configure/enable the sandbox using the Edit Problem dialog (or via YAML configuration).
+     * 
+     * @param sandboxProgramName the name of the sandbox program to be used by this Problem, when sandbox usage is enabled.
+     */
+    public void setSandboxProgramName(String sandboxProgram) {
+        this.sandboxProgramName = sandboxProgram;
+    }
+
+    /**
+     * Returns the String containing the command used to invoke the sandbox configured for this problem.
+     * Note that the returned value is meaningless if the Problem has not been configured to use a sandbox. 
+     * 
+     * @return the command line used to invoke the sandbox for this problem, when sandbox usage is enabled.
+     */
+    public String getSandboxCmdLine() {
+        if (sandboxCmdLine == null) {
+            sandboxCmdLine = Constants.PC2_INTERNAL_SANDBOX_COMMAND_LINE;
+        }
+        return sandboxCmdLine;
+    }
+
+    /**
+     * Sets the command line used to invoke the sandbox associated with this Problem.
+     * Note that setting the sandbox command line does not in and of itself enable the use of a sandbox; the
+     * Admin must enable the sandbox via the Edit Problem dialog (or via YAML configuration).
+     * Note also that the value of sandboxCmdLine is meaningless if the Problem is currently configured
+     * with {@link SandboxType#NONE}.
+     * 
+     * @param sandboxCmdLine the command line used to invoke the Problem sandbox.
+     */
+    public void setSandboxCmdLine(String sandboxCmdLine) {
+        this.sandboxCmdLine = sandboxCmdLine;
+    }
+
+    /**
+     * Returns a boolean flag which indicates whether this Problem has been configured to use a sandbox.
+     *
+     * @return false if the currently configured SandboxType for the problem is {@link Problem.SandboxType#NONE}; 
+     *          true if any other sandbox type has been configured.
+     */
+    public boolean isUsingSandbox() {
+        return sandboxType != SandboxType.NONE;
+    }
+
+    /**
+     * Returns the type of sandbox configured in this Problem; an element of {@link Problem.SandboxType} 
+     * which might be {@link SandboxType#NONE}.
+     * 
+     * @return an element of {@link Problem.SandboxType}.
+     */
+    public SandboxType getSandboxType() {
+        if (sandboxType == null) {
+            sandboxType = SandboxType.NONE;
```

Comment:
Same comment as for methods `getSandboxProgramName()` and `getSandboxCmdLine()` (above):  I don't think a getter should be making a side-effect change.  Someone *slap* the programmer who originally wrote this code :)

Reply:
I don't think this one is a problem.  `null `really implies that it is type `NONE `since really, `SandboxType `is an _enum_, not an object (really).

Reply:
Ok, I'll buy that.